### PR TITLE
Fix for CSG nodes not being able to be exported as glTF

### LINKED ISSUE #57667: CSG export to gltf failed
### Godot version

4.0 alpha 1

### System information

popos 21.04 Linux, Vulkan, NVIDIA 3000 series

### Issue description

Exporting CSG as gltf fails.

### Steps to reproduce

1. Create a first csg box
2. Create a second csg box
3. Change the second box to be intersect
4. Export as gltf
5. gltf can not be read in godot or using a gltf validator or blender 3

### Minimal reproduction project

[CSG Export Game Project.zip](https://github.com/godotengine/godot/files/8008427/CSG.Export.Game.Project.zip)

### PR DESCRIPTION
**Pull request by fazil47**
This is an attempt to fix #57667

image

It works but unlike `GLTFDocument::_convert_mesh_to_gltf` when a material isn't applied, I've assigned a default `StandardMaterial3D` because of #57955

*Bugsquad edit:*
- Fixes #57667

### CODE REVIEW

File: modules/gltf/gltf_document.cpp [hunk 1]
```diff
@@ -5322,14 +5322,31 @@ void GLTFDocument::_convert_csg_shape_to_gltf(CSGShape3D *p_current, GLTFNodeInd
...
+
+		for (int32_t surface_i = 0; surface_i < csg_mesh->get_surface_count(); surface_i++) {
+			Array array = csg_mesh->surface_get_arrays(surface_i);
+			Ref<Material> mat = csg_mesh->surface_get_material(surface_i);
+			String mat_name;
+			if (mat.is_valid()) {
+				mat_name = mat->get_name();
+			} else {
+				// Assign default material when no material is assigned.
+				mat = Ref<StandardMaterial3D>(memnew(StandardMaterial3D));
```
Comment (fire) [inline modules/gltf/gltf_document.cpp:5339]: I think `mat = memnew(StandardMaterial3D);` will work too.

**Comment by fire** [comment]
I tested using https://github.khronos.org/glTF-Validator/ and with a test project.
[3d csg export Game Project.zip](https://github.com/godotengine/godot/files/8048914/3d.csg.export.Game.Project.zip)

Seems to work.

**Review by fire — APPROVED** [review]
This pr solves the CSG gltf export problem, there's minor style differences but it doesn't affect the testing results.

**Review by akien-mga — APPROVED** [review]

File: modules/gltf/gltf_document.cpp (hunk 1 above)
Comment (fire) [reply modules/gltf/gltf_document.cpp:5339]: Makes sense.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**